internal/static/s3/minio.go

```diff
@@ -0,0 +1,143 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+
+	"github.com/caos/logging"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+
+	"github.com/caos/zitadel/internal/domain"
+	caos_errs "github.com/caos/zitadel/internal/errors"
+)
+
+type Minio struct {
+	Client   *minio.Client
+	Location string
+}
+
+func NewMinio(config S3Config) (*Minio, error) {
+	minioClient, err := minio.New(config.Endpoint, &minio.Options{
+		Creds:  credentials.NewStaticV4(config.AccessKeyID, config.SecretAccessKey, ""),
+		Secure: config.SSL,
+		Region: config.Location,
+	})
+	if err != nil {
```

Comment:
IMHO we should wrap the errors with the caos error pkg so we don't have to handle specific errors in packages using the static package

Reply:
like
```
return caos_errs.ThrowAlreadyExists(nil, "MINIO-9n3MK", "Errors.Assets.Bucket.AlreadyExists")
```